code/modules/mob/mob.dm

```diff
@@ -935,7 +935,7 @@
 		return TRUE
 
 	var/has_casting_restrictions = SEND_SIGNAL(src, COMSIG_MOB_RESTRICT_MAGIC, src, magic_flags)
-	return has_casting_restrictions
+	return !has_casting_restrictions
```

Comment:
Don't do this, signals return flags, not booleans.

Change the variable name on this to `restrict_magic_flags` or similar, and this to `return !(restrict_magic_flags & WHATEVER_FLAG_YOU_WANT)`

Reply:
Applied with https://github.com/tgstation/tgstation/pull/64124/commits/3a628c81f72336a63e390bd0e051a24c1c3e9124